src/System.Security.Cryptography.Xml/tests/CipherDataTests.cs

```diff
@@ -0,0 +1,206 @@
+                    xmlElement.OuterXml);
+            }
+        }
+
+        public static IEnumerable<object[]> Constructor_CipherReference_Source()
+        {
+            return new object[][]
+            {
+                new [] { new CipherReference() },
+                new [] { new CipherReference("http://dummy.urionly.io") },
+                new [] { new CipherReference("http://dummy.uri.transform.io", new TransformChain()) },
+            };
+        }
+
+        [Fact]
+        public void CipherReference_Null()
+        {
+            CipherData cipherData = new CipherData();
+            Assert.Throws<ArgumentNullException>(() => cipherData.CipherReference = null);
+        }
+
+        [Fact]
+        public void CipherReference_CipherValueSet()
+        {
+            CipherData cipherData = new CipherData(new byte[0]);
+            Assert.Throws<CryptographicException>(() => cipherData.CipherReference = new CipherReference());
+        }
+
+        [Fact]
+        public void CipherValue_Null()
+        {
+            CipherData cipherData = new CipherData(new CipherReference());
+            Assert.Throws<ArgumentNullException>(() => cipherData.CipherValue = null);
+        }
+
+        [Fact]
+        public void CipherValue_CipherReferenceSet()
+        {
+            CipherData cipherData = new CipherData(new CipherReference());
+            Assert.Throws<CryptographicException>(() => cipherData.CipherValue = new byte[0]);
+        }
+
+        [Fact]
+        public void LoadXml_Null()
+        {
+            CipherData cipherData = new CipherData();
+            Assert.Throws<ArgumentNullException>(() => cipherData.LoadXml(null));
+        }
+
+        [Fact]
+        public void LoadXml_NoValueOrReference()
+        {
+            XmlDocument xmlDocument = new XmlDocument();
+            xmlDocument.LoadXml("<root/>");
+
+            CipherData cipherData = new CipherData();
+
+            Assert.Throws<CryptographicException>(() => cipherData.LoadXml(xmlDocument.DocumentElement));
+        }
+
+        [Theory]
+        [MemberData(nameof(LoadXml_CipherValue_Source))]
+        public void LoadXml_CipherValue(XmlElement xmlElement, byte[] cipherValue)
+        {
+            CipherData cipherData = new CipherData();
+            cipherData.LoadXml(xmlElement);
+
+            Assert.Equal(cipherValue, cipherData.CipherValue);
+            Assert.Null(cipherData.CipherReference);
+
+            XmlElement gotXmlElement = cipherData.GetXml();
+            Assert.NotNull(gotXmlElement);
+            Assert.Equal(xmlElement.OuterXml, gotXmlElement.OuterXml);
+        }
+
+        public static IEnumerable<object[]> LoadXml_CipherValue_Source()
+        {
+            return new []
+            {
+                ToCipherDataTestCase("<root xmlns:enc='{0}'><enc:CipherValue>{1}</enc:CipherValue></root>", new byte[0]),
```

Comment:
Thank you. I like to put a bit of effort into making tests easy to write and expand.